nvtabular/io.py

```diff
@@ -331,123 +332,178 @@ def __iter__(self):
             yield from GPUFileIterator(path, **self.kwargs)
 
 
-class Shuffler:
-    """
-    Shuffling the data is an important part of machine learning
-    training. This class is used by Workflow class and shuffles
-    the data after all the pre-processing and feature engineering
-    operators are finished their processing.
+class Writer:
+    def __init__(
+        self,
+        out_dir,
+        num_out_files=30,
+        num_threads=4,
+        output_format="parquet",
+        cats=None,
+        conts=None,
+        labels=None,
+    ):
+        # set variables
+        self.num_out_files = num_out_files
+        self.writer = None
+        if output_format == "binary":
+            self.writer = HugeCTRWriter(out_dir, num_out_files, num_threads, cats, conts, labels)
+        elif output_format == "parquet":
+            self.writer = ParquetFWriter(out_dir, num_out_files, num_threads, cats, conts, labels)
 
-    Parameters
-    -----------
-    out_dir : str
-        path for the shuffled files
-    num_out_files : int, default 30
-    num_threads : int, default 4
+    @annotate("add_data", color="orange", domain="nvt_python")
+    def add_data(self, gdf):
+        arr = cp.arange(len(gdf))
+        b_idxs = np.arange(self.num_out_files)
+        self.writer.write_data(gdf, arr, b_idxs)
```

Comment:
I think the body of this method should be ```raise NotImplementedError``` - and we should be requiring the subclasses to override.

```suggestion
    def add_data(self, gdf):
        raise NotImplementedError()
```